src/main/java/be/syntra/devshop/DevshopFront/services/SearchServiceImpl.java

```diff
@@ -60,4 +66,26 @@ public SearchModel getSearchModel() {
     public void resetSearchModel() {
         this.searchModel = new SearchModel();
     }
+
+    @Override
+    public void setAppliedFiltersToSearchModel() {
+        searchModel.setAppliedFiltersHeader(" with the applied filters");
+        String searchRequest = hasSearchRequest()
+                ? getSearchModel().getSearchRequest()
+                : "";
+        searchModel.setSearchRequest(searchRequest);
+        searchModel.setSearchFailure(false);
+        searchModel.setActiveFilters(true);
+    }
+
+    private boolean hasSearchRequest() {
+        return null != getSearchModel().getSearchRequest();
+    }
+
+    @Override
+    public void setPriceFilters(List<Product> products) {
+        BigDecimal priceHigh = products.stream().map(Product::getPrice).max(Comparator.naturalOrder()).orElse(BigDecimal.ZERO);
```

Comment:
try to make a habit of splitsing a stream over multiple lines. 
products.stream()
.map
.max
.orElse

Reply:
split